speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl

```diff
@@ -55,17 +55,18 @@ void DecisionMaker<SocketBlackBox, SDAConfig>::Initialize(tCarElt* p_initialCar,
     BlackBox.Initialize(initialData, p_testSituations, p_testAmount);
 
     std::experimental::filesystem::path blackBoxPath = std::experimental::filesystem::path(p_blackBoxExecutablePath);
-    m_fileBufferStorage.Initialize(Config.GetDataCollectionSetting(),
-                                   BUFFER_FILE_PATH,
-                                   Config.GetUserId(),
-                                   std::chrono::system_clock::to_time_t(std::chrono::system_clock::now()),
-                                   blackBoxPath.filename().string(),
-                                   blackBoxPath.stem().string(),
-                                   std::chrono::system_clock::to_time_t(std::experimental::filesystem::last_write_time(blackBoxPath)),
-                                   p_track->filename,
-                                   p_track->name,
-                                   p_track->version,
-                                   Config.GetInterventionType());
+    m_bufferFilePath = m_fileBufferStorage.Initialize(
+        Config.GetDataCollectionSetting(),
+        BUFFER_FILE_PATH,
```

Comment:
The value of BUFFER_FILE_PATH should also be updated now that the temporary directory is properly used